src/test/ui/write-to-static-mut-in-static.stderr

```diff
@@ -5,10 +5,10 @@ LL | pub static mut B: () = unsafe { A = 1; };
    |                                 ^^^^^ modifying a static's initial value from another static's initializer
 
 error[E0391]: cycle detected when const-evaluating `C`
-  --> $DIR/write-to-static-mut-in-static.rs:5:34
+  --> $DIR/write-to-static-mut-in-static.rs:5:1
    |
 LL | pub static mut C: u32 = unsafe { C = 1; 0 };
-   |                                  ^^^^^
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
what is that span?! From the start of `pub` to the end of `unsafe`?

anyway, this should only be pointing at the `C` in fact, not even the `C = 1`

Reply:
> what is that span?! From the start of pub to the end of unsafe?

I seem to recall us having some heuristics to find the "def span" which involves looking for the first `{`...

Reply:
> anyway, this should only be pointing at the C in fact, not even the C = 1

Probably, `C = 1;` is a single MIR statement, so it's not clear how to be more precise than that.

Reply:
This is certainly caused by some bad use of `guess_head_span`. I've been removing some of them, but we need a thorough audit of its usages.